### PR TITLE
Freeradius upgrade and bugfix

### PR DESCRIPTION
**Pull request by geaaru**
Finally, I push a lot of fixes for open issues and for bump a new release.

### CODE REVIEW

**Comment by jsmolic** [comment]
I suppose you meant to write `Closes: https://bugs.gentoo.org/${BUG_NUMBER}` rather than\
`Closes: https://github.com/gentoo/gentoo/pull/${BUG_NUMBER}` in the commit messages to close the bugs...

**Comment by geaaru** [comment]
Yeah, sorry. Typo. I will fix this.

**Comment by jsmolic** [comment]
> Yeah, sorry. Typo. I will fix this.

You can also use e.g. `repoman -c 725336` and `Closes:  https://bugs.gentoo.org/725336` will be automatically added to your commit message ;)

**Comment by geaaru** [comment]
Ah cool, i note this for the next time. Thanks for the tips.

**Comment by jsmolic** [comment]
@geaaru I see that #630910 is a security bug, you should probably use tag `Bug: ` instead of `Closes: ` for that one, Gentoo security team likes to use those bugs for cleanup, stabilization, etc. and they close off the bugs afterwards.\
You can also append `[please reassign]` to the PR title, then the bot will link the PR in the bug and security team will be notified of this PR so it will be reviewed more promptly.

**Comment by geaaru** [comment]
> @geaaru I see that #630910 is a security bug, you should probably use tag `Bug: ` instead of `Closes: ` for that one, Gentoo security team likes to use those bugs for cleanup, stabilization, etc. and they close off the bugs afterwards.
> You can also append `[please reassign]` to the PR title, then the bot will link the PR in the bug and security team will be notified of this PR so it will be reviewed more promptly.

Ok. Done. Thanks

File: net-dialup/freeradius/freeradius-3.0.20.ebuild [hunk 1]
```diff
@@ -242,10 +246,10 @@ src_install() {
 			-e 's:^WatchdogSec=.*::g' -e 's:^NotifyAccess=all.*::g' \
 			"${S}"/debian/freeradius.service
 	fi
-	systemd_newtmpfilesd "${FILESDIR}"/freeradius.tmpfiles freeradius.conf
```
[the remark this answers is not in the document]
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.20.ebuild:245]: This command is a legacy command of the previous maintainer. It simply fixes the permissions that are handled directly from systemd with User/Group option. It's not needed. Just drop it.

File: net-dialup/freeradius/freeradius-3.0.20.ebuild [hunk 2]
```diff
@@ -228,6 +228,10 @@ src_install() {
 
 	pamd_mimic_system radiusd auth account password session
 
+	# fix #711756
+	fowners -R radius:radius /etc/raddb
+	fowners -R radius:radius /var/log/radius
```
[the remark this answers is not in the document]
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.20.ebuild:233]: Please, this is not on the post_install phase but it's on the install phase inside the sandbox. It's safe.

See this: https://bugs.gentoo.org/711756

File: net-dialup/freeradius/freeradius-3.0.21.ebuild [hunk 3]
```diff
@@ -0,0 +1,266 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
```
[the remark this answers is not in the document]
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.21.ebuild:6]: Next commit, of course. But please close this issue for now.

File: net-dialup/freeradius/freeradius-3.0.20.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-dialup/freeradius/freeradius-3.0.20.ebuild:233]: I'm not sure if I'm misunderstanding here. It's not about safe or unsafe, it's about how we want to ensure that users update and get the fixed version - hence a [revision](https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html) bump, no?

File: net-dialup/freeradius/freeradius-3.0.20.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-dialup/freeradius/freeradius-3.0.20.ebuild:245]: Sure.

File: net-dialup/freeradius/freeradius-3.0.21.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (thesamesam) [reply net-dialup/freeradius/freeradius-3.0.21.ebuild:6]: That's why I said next time 👍

File: net-dialup/freeradius/freeradius-3.0.20.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.20.ebuild:233]: Ah, it's true. My fault. I will bump 3.0.20-r1

File: net-dialup/freeradius/freeradius-3.0.21.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.21.ebuild:6]: I will check the support. Not sure if the py3.9 has broken API changes on C integration.

File: net-dialup/freeradius/freeradius-3.0.21.ebuild [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+}
+
+pkg_config() {
+	if use ssl; then
+		cd "${ROOT}"/etc/raddb/certs || die
+		./bootstrap || die "Error while running ./bootstrap script."
+		fowners root:radius "${ROOT}"/etc/raddb/certs
+		fowners root:radius "${ROOT}"/etc/raddb/certs/ca.pem
+		fowners root:radius "${ROOT}"/etc/raddb/certs/server.{key,crt,pem}
```
Comment (juippis) [inline net-dialup/freeradius/freeradius-3.0.21.ebuild:257]: I have a feeling `fowners` won't work in `pkg_config`. Could you test this? Replace with `chown ... || die` if not.

Comment (geaaru) [reply net-dialup/freeradius/freeradius-3.0.21.ebuild:257]: Tested in my overlay: https://github.com/geaaru/geaaru_overlay/commit/4a8098530946f27feebece37610166cf4c6ad2ce
`fowners` works correctly.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**